File: lesson6/python3/homework/homework6_4.py
#!/usr/bin/python3
#: lesson6/python3/homework/homework6_4.py


def greatest_common_factor(numb_x, numb_y):
    numb_res = numb_x + numb_y
    count = 1
    res = 0
    while count <= numb_res:
        if numb_res % count == 0:
            if res < count:
                res = count
        count += 1
    return res


numb1 = int(input("Введите число -> "))
numb2 = int(input("Введите число -> "))

result = greatest_common_factor(numb1, numb2)
print('result ->', result)

File: lesson6/c_c++/homework/homework6_4.c
/*
 * lesson6/c_c++/homework/homework6_4.c
 * Напишите функцию, который вычисляет наибольший общий делитель делитель.
 */
#include <stdio.h>

int greatestCommonFactor(int numb_x, int numb_y) {
    int numb_res = numb_x + numb_y;
    int res = 0;
    int count = 1;
    while (count <= numb_res) {
        if (numb_res % count == 0) {
           if (res < count) {
               res = count;
           }
        }
        count++;
    }
    return res;
}

int main(int argc, char** argv) {
    int numb1, numb2;
    printf("Введите первое число -> ");
    scanf("%d", &numb1);
    printf("Введите второе число -> ");
    scanf("%d", &numb2);
    int numb_result = greatestCommonFactor(numb1, numb2);
    printf("Наибольший общий делитель -> %d\n", numb_result);
    return 0;
}
